src/Cli/dotnet/commands/dotnet-run/LaunchSettings/LaunchSettingsManager.cs

```diff
@@ -50,9 +50,27 @@ public static LaunchSettingsApplyResult TryApplyLaunchSettings(string launchSett
                             .EnumerateObject()
                             .FirstOrDefault(IsDefaultProfileType).Value;
                     }
-                    else
+                    else // Find a profile match for the given profileName
                     {
-                        if (!profilesObject.TryGetProperty(profileName, out profileObject) || profileObject.ValueKind != JsonValueKind.Object)
+                        IEnumerable<JsonProperty> caseInsensitiveProfileMatches = profilesObject
+                            .EnumerateObject() // p.Name shouldn't fail, as profileObject enumerables here are only created from an existing JsonObject
+                            .Where(p => string.Compare(p.Name, profileName, StringComparison.OrdinalIgnoreCase) == 0);
```

Comment:
Nit: I think `Equals` is easier to read than `Compare`.

Also, it's probably a good idea to add a `ToList()` to the end of this so that the enumeration / comparison doesn't have to run multiple times (for the `Count()` call, then the `Any()` call, then the `First()` call).

```suggestion
                            .Where(p => string.Equals(p.Name, profileName, StringComparison.OrdinalIgnoreCase))
                            .ToList();
```